### DISCUSSION TITLE
Jackson serialization - unable to deserialize scala.util.Either from remote actor response

### DISCUSSION THREAD

**Discussion by farooqkamal** [apache/pekko#1387] [Q&A]
Hi Folks,

For a response from a remote actor in the same cluster, the standard java serialization works fine for `Either`, whereas using `jackson` (CBOR or  Json) converts the Either response to a HashMap and obviously does not deserialize. Wondering if someone has faced this and has a solution. Below is the exception

```
ClassCastException: class scala.collection.immutable.HashMap cannot be cast to class x.y.MyClass (scala.collection.immutable.HashMap is in unnamed module of loader play.runsupport.NamedURLClassLoader @5686bcf7; x.y.MyClass is in unnamed module of loader play.runsupport.DelegatedResourcesClassLoader @28a7de20
```

**Comment by pjfanning** [comment]
You appear to be using Play. I would suggest that you read the Play docs relating to serialization. Play 3 does use Pekko under the hood but I am not clear from your description that you have provided whether you are using pekko-serialization-jackson directly or indirectly.

pekko-serialization-jackson is documented here:
https://pekko.apache.org/docs/pekko/current/serialization-jackson.html

**Comment by farooqkamal** [comment]
Yes I'm using Play 3 with the preconfigured serialization settings

```
pekko.actor {
  serializers {
    jackson-json = "org.apache.pekko.serialization.jackson.JacksonJsonSerializer"
    jackson-cbor = "org.apache.pekko.serialization.jackson.JacksonCborSerializer"
  }
   serialization-bindings {
      "x.y.CborSerializable" = jackson-cbor // marker trait
      "scala.util.Either" = jackson-cbor
      // "scala.util.Right" = jackson-cbor - tried just in case but has no effect
    }
}
``` 
If the response is a simple case class with primitives e.g. `MyCaseClass extends CborSerializable` it works fine, only when the response is `Either[MyError, MyCaseClass]` it fails to correctly serialize
